app/assets/stylesheets/header.scss

```diff
@@ -145,4 +116,97 @@
       margin-right: 10px;
     }
   }
+
+  @media (max-width: $grid-float-breakpoint-max) {
+
+    .navbar-nav.hidden-xs {
+      margin: 0;
+      margin-top: 6px;
+    }
+
+    .nav-badge {
+      color: $navbar-inverse-link-color;
+      &:hover { color: $navbar-inverse-link-hover-color; }
+      &:hover,
+      &:focus {
+        background-color: transparent;
+      }
+    }
+
+    #navbar-collapse {
+      .form-group,
+      .twitter-typeahead {
+        display: block !important;
+        margin-bottom: 0;
+
+        &,
+        input { width: 100%; }
+      }
+    }
+
+    .nav-badges .dropdown-menu {
+      width: 300px;
+    }
+
+    #user_menu.dropdown-menu {
+      display: block;
+      position: static;
+      width: 100%;
+      background-color: transparent;
+      box-shadow: none;
+      border: 0;
+      padding: 0;
+      margin-top: -8px; // To compensate parent ul margin
+
+      a {
+        color: #9d9d9d;
+        font-weight: bold;
+        padding: 10px 15px;
+
+        &:hover {
+          color: #fff;
+          background-color: transparent;
+        }
+      }
+    }
+  }
+
+  @media (min-width: $grid-float-breakpoint) {
+
+    [type="search"] {
+      @include transition(width);
+      margin-top: 2px;
+      width: 200px;
+
+      &:not(.active) {
+        background-color: $navbar-inverse-bg;
+        border-color: $gray-light;
+        width: 150px;
+      }
+    }
+
+    #user_menu {
+      &.open .dropdown-toggle { background-color: darken($navbar-inverse-bg, 7%); }
+      .dropdown-toggle {
+        margin: 0 1px;
+        min-width: 160px;
+      }
+
+      .dropdown-menu {
+        background-color: darken($navbar-inverse-bg, 7%);
+        border-top: 0;
+        width: 100%;
+
+        > li > a {
```

Comment:
- Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 5